tests/tests/test_state_scripts.py

```diff
@@ -873,6 +873,147 @@ def fetch_info(cmd_list):
+            "ScriptOrder": [
+                "Idle_Enter_08_testing",
+                "Idle_Enter_09",
+                "Idle_Leave_09",
+                "Idle_Leave_10",
+                "Sync_Enter_02",
+                "Sync_Enter_03",
+                "Sync_Leave_04",
+                "Sync_Leave_15",
+                "Download_Enter_12",
+                "Download_Enter_13",
+                "Download_Leave_14",
+                "Download_Leave_25",
+                "ArtifactInstall_Enter_01",
+                "ArtifactInstall_Enter_02",
+                "ArtifactInstall_Leave_01",
+                "ArtifactInstall_Leave_03",
+                "ArtifactRollback_Enter_01",
+                "ArtifactRollback_Enter_11",
+                "ArtifactRollback_Leave_01",
+                "ArtifactRollback_Leave_89",
+                "ArtifactFailure_Enter_01",
+                "ArtifactFailure_Leave_06",
+                "Idle_Enter_08_testing",  # Update attempt finished
+                "Idle_Enter_09",
+            ],
+        },
+    )
+
+    @MenderTesting.slow
+    @pytest.mark.parametrize("description,test_set", TEST_SET_ROLLBACK)
+    def test_rollback_on_current_active_partition(
+            self, standard_setup_one_client_bootstrapped, valid_image, description, test_set
+    ):
+        """Test that when the client performs a rollback correctly on
+ArtifactInstall_Leave errors"""
+
+        mender_device = standard_setup_one_client_bootstrapped.device
+        work_dir = "test_state_scripts.%s" % mender_device.host_string
+
+        script_content = (
+            '#!/bin/sh\n\necho "$(basename $0)" >> /data/test_state_scripts.log\n'
+        )
+
+        script_failure_content = script_content + "exit 1\n"
+
+        artifact_script_dir = os.path.join(work_dir, "artifact-scripts")
+
+        if os.path.exists(work_dir):
+            shutil.rmtree(work_dir, ignore_errors=True)
+
+        os.mkdir(work_dir)
+        os.mkdir(artifact_script_dir)
+
+        new_rootfs = os.path.join(work_dir, "rootfs.ext4")
+        shutil.copy(valid_image, new_rootfs)
+
+        ps = subprocess.Popen(["debugfs", "-w", new_rootfs], stdin=subprocess.PIPE)
+        ps.stdin.write(b"cd /etc/mender\n" b"mkdir scripts\n" b"cd scripts\n")
+        ps.stdin.close()
+        ps.wait()
+
+        for script in test_set.get("ScriptOrder"):
+            if not script.startswith("Artifact"):
+                # Not an artifact script, skip this one.
+                continue
+            with open(os.path.join(artifact_script_dir, script), "w") as fd:
+                if script in test_set.get("FailureScript", []):
+                    fd.write(script_failure_content)
+                else:
+                    fd.write(script_content)
+
+        # Create the artifact, and make the deployment.
+        device_id = Helpers.ip_to_device_id_map(
+            MenderDeviceGroup([mender_device.host_string])
+        )[mender_device.host_string]
```

Comment:
I wonder why we need this. It is in all the three tests and probably just a leftover since before the Python 3 refactoring, but I don't think it is necessary.

Passing `devices=None` (default) to `common_update_procedure` should have the same effect, right? Do you dare to try? :)